### DISCUSSION TITLE
Is there a way to format only the modified parts of a file?

### DISCUSSION THREAD

**Discussion by rohitshriwas** [zed-industries/zed#16342] [Feature Requests]
The format_on_save feature is very nice and it can be nicer if we can get it to format only the parts of a file that were modified since the last git commit. Is this feature already present? Can somebody please confirm?

Here is a screenshot of how it looks on vscode for reference:
![image](https://github.com/user-attachments/assets/f36e2edd-7c68-4285-8b7e-12bd5638e36d)

**Comment by pjv** [comment]
Follow #16509

**Reply by rohitshriwas** [reply]
It's a critical issue indeed. Thank you 😊
